src/CoreWCF.WebHttp/src/CoreWCF/OpenApi/OpenApiSchemaBuilder.cs

```diff
@@ -0,0 +1,631 @@
+                            definitionSchema.Properties.Add(name, new OpenApiDefinitionValueProperty
+                            {
+                                Type = "array",
+                                Description = memberPropertiesAttribute?.Description,
+                                Items = new OpenApiDefinitionValueProperty
+                                {
+                                    Reference = $"#/components/schemas/{innerDataMemberAttribute.Name}"
+                                }
+                            });
+                        }
+                        else
+                        {
+                            string openApiType = GetType(innerType);
+
+                            if (!string.IsNullOrEmpty(openApiType))
+                            {
+                                definitionSchema.Properties.Add(name, new OpenApiDefinitionValueProperty
+                                {
+                                    Type = "array",
+                                    Description = memberPropertiesAttribute?.Description,
+                                    Items = new OpenApiDefinitionValueProperty
+                                    {
+                                        Type = openApiType,
+                                    }
+                                });
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    definitionSchema.Properties.Add(name, new OpenApiDefinitionValueProperty
+                    {
+                        Type = GetType(property.PropertyType),
+                        Description = memberPropertiesAttribute?.Description,
+                    });
+                }
+            }
+
+            definitionSchema.Required = required.Count > 0 ? required : null;
+
+            schema.Components.Schemas.Add(dataContractAttribute.Name, definitionSchema);
+        }
+
+        /// <summary>
+        /// Map a .NET type to JSON schema type.
+        /// </summary>
+        /// <param name="type">The type to be mapped.</param>
+        /// <returns>The mapped type.</returns>
+        private static string GetType(Type type)
+        {
+            if (type == null)
+            {
+                return null;
+            }
+
+            Type actualType = IsNullable(type) ? Nullable.GetUnderlyingType(type) : type;
+
+            if (actualType == typeof(int) || actualType == typeof(long) || actualType == typeof(short) || actualType == typeof(byte))
+            {
+                return "integer";
+            }
+            else if (actualType == typeof(float) || actualType == typeof(double) || actualType == typeof(decimal))
+            {
+                return "number";
+            }
+            else if (actualType == typeof(string) || actualType == typeof(DateTime) || actualType == typeof(Stream) || actualType == typeof(Guid))
+            {
```

Comment:
Should you include DateTimeOffset in this list? Where does char fit in? I think it might fit here too if there's the ability to constrain the length to 1.

Reply:
This is addressed.